src/editable.js

```diff
@@ -0,0 +1,191 @@
+/**
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/* globals HTMLTextAreaElement */
+
+'use strict';
+
+/**
+ * Represents an editable inside the editor.
+ *
+ * @class Editable
+ * @extends Model
+ */
+
+CKEDITOR.define( [ 'model', 'config', 'editablecollection', 'promise' ], function( Model, Config, EditableCollection, Promise ) {
+	var Editable = Model.extend( {
+		/**
+		 * Creates an instance of the Editable class.
+		 *
+		 * @constructor
+		 * @param {HTMLElement|Editable} element The DOM element that will be linked to this editable. If an editable is
+		 * provided, that same editable is returned by the constructor.
+		 */
+		constructor: function Editable( element ) {
+			// We need to renew the module request because of cross-reference with Editable<->EditableCollection.
+			EditableCollection = CKEDITOR.require( 'editablecollection' );
+
+			// Accept and Editable instance as parameter as well, for simplification in other parts of the code.
+			if ( element instanceof Editable ) {
+				return element;
+			}
+
+			// Call the base constructor.
+			Model.apply( this );
+
+			/**
+			 * The DOM element managed by this editable.
+			 *
+			 * @readonly
+			 * @property {HTMLElement} element
+			 */
+			this.set( 'element', element );
+
+			/**
+			 * The DOM element that represents the editable view.
+			 *
+			 * @readonly
+			 * @property {HTMLElement} view
+			 */
+			this.set( 'view' );
+
+			/**
+			 * Configurations specific to this editable. It inherits configurations from parent editables.
+			 *
+			 * @readonly
+			 * @property {Config} config
+			 */
+			this.set( 'config', new Config() );
+
+			/**
+			 * The child editables of this editable.
+			 *
+			 * @readonly
+			 * @property {EditableCollection} editables
+			 */
+			this.set( 'editables', new EditableCollection( this ) );
+
+			return this;
+		},
+
+		/**
+		 * Initializes the editable.
+		 *
+		 * The initialization consists of the following procedures:
+		 *
+		 *  * Setup the editable view.
+		 *  * Initialize children.
+		 *  * Enable editing in the editable.
+		 *
+		 * This method should be rarely used as `editor.init` calls it during editor initialization.
+		 *
+		 * @returns {Promise} A promise that resolves once the initialization is completed.
+		 */
+		init: function() {
+			var that = this;
+
+			this._initPromise = this._initPromise || Promise.resolve()
```

Comment:
There is missing test for it. I can remove `this._initPromise ||` and tests are still green.